### PR TITLE
Feature/issue 299 add multisite functionality

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

**Comment by codiebeulaine** [comment]
@miltontony ready for review

File: molo/core/models.py [hunk 1]
```diff
@@ -331,6 +331,42 @@ def move(self, target, pos=None):
...
+                    is_active=False)
+            else:
+                new_lang = destination_site.languages.languages.filter(
+                    locale=language.language.locale).first()
+            return new_lang
+
+    def copy(self, *args, **kwargs):
+        current_site = self.get_site()
+        destination_site = kwargs['to'].get_site()
+        if not (current_site is destination_site):
```
Comment (miltontony) [inline molo/core/models.py]: This reads better for me `if current_site is not destination_site:` (without the brackets)

File: molo/core/models.py [hunk 2]
```diff
@@ -331,6 +331,42 @@ def move(self, target, pos=None):
...
+            new_lang = self.copy_languages(current_site, destination_site)
+            page_copy = super(TranslatablePageMixin, self).copy(
+                *args, **kwargs)
+            new_l_rel = LanguageRelation.objects.get(page=page_copy)
```
Comment (miltontony) [inline molo/core/models.py]: Let's make this `get_or_create` just to be safe

File: molo/core/wagtail_hooks.py [hunk 3]
```diff
@@ -37,12 +37,40 @@ def register_admin_urls():
...
     if main_language and parent_page.depth > 2:
         return pages.filter(languages__language__locale=main_language.locale)
     return pages
 
 
+@hooks.register('after_copy_page')
+def copy_translation_pages(request, page, new_page):
+    current_site = page.get_site()
+    destination_site = new_page.get_site()
+    if not (current_site is destination_site):
```
Comment (miltontony) [inline molo/core/wagtail_hooks.py]: Same here for style

File: molo/core/wagtail_hooks.py [hunk 4]
```diff
@@ -37,12 +37,40 @@ def register_admin_urls():
...
+        page.specific.copy_languages(current_site, destination_site)
+    languages = Languages.for_site(destination_site).languages
+    if (languages.filter(
+            is_main_language=True).exists() and
+            not new_page.languages.exists()):
```
Comment (miltontony) [inline molo/core/wagtail_hooks.py]: I think this `if` can fit in `2` lines instead of `3`

**Pull request merged**